src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -777,6 +775,66 @@ public int EnsureCapacity(int capacity)
             return newSize;
         }
 
+        /// <summary>
+        /// Sets the capacity of this dictionary to hold up at least all existing entries
+        /// 
+        /// This method can be used to minimize the memory overhead 
+        /// once it is known that no new elements will be added. 
+        /// 
+        /// To completely clear a dictionary and release all memory 
+        /// referenced by the dictionary, execute the following statements:
+        /// 
+        /// dictionary.Clear();
+        /// dictionary.TrimExcess();
+        /// </summary>
+        public void TrimExcess()
+        {
+            TrimExcess(HashHelpers.GetPrime(Count));
+        }
+
+        /// <summary>
+        /// Sets the capacity of this dictionary to hold up at least 'capacity' entries without any further expansion of its backing storage
+        /// </summary>
+        public void TrimExcess(int capacity)
+        {
+            if (capacity < Count)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.capacity);
+            }
+
+            if (_entries != null && capacity < _entries.Length)
```

Comment:
@jkotas from a CQ perspective, as a rule of thumb is early return better, worse, or not generally one way or another?

That aside, I generally prefer to early out when the rest of the method would be enclosed by the if, as it's easier to follow and indents less.

Reply:
It should not matter in this case. I think that the JIT will generate identical code for it because of we have started folding the epilogs recently.

There is a much more fundamental code size issue in this delta in that it introduces another copy of the rehashing loop.